db/models/chunk.js

```diff
@@ -1,242 +1,35 @@
-                return await StorageLink.byChunkIdAndStatus(this.id, status);
-            },
-            set: async(x, status, value) => {
-                // todo
-            }
-        });
-    }
-
-    // todo: delete
-    // async getStorageLinkIds(status = null) {
-    //     if (! status) status = Chunk.STORAGE_LINK_STATUS_ALL;
-    //     if (! this.storage_link_ids) return [];
-    //     if (! this.storage_link_ids[status]) return [];
-    //     if (status === Chunk.STORAGE_LINK_STATUS_ALL) {
-    //         return (await Promise.all(Chunk.STORAGE_LINK_STATUSES.map(async(status) => await this.getStorageLinkIds(status)))).flat();
-    //     }
-    //     return this.storage_link_ids[status];
-    // }
-
-    static getChunkStoragePath(id) {
-        const cache_dir = path.join(this.ctx.datadir, this.ctx.config.client.storage.cache_path);
-        utils.makeSurePathExists(cache_dir);
-        return path.join(cache_dir, 'chunk_' + id);
-    }
-
-    static forceSaveToDisk(data, id = null) {
-        if (!Buffer.isBuffer(data)) throw Error('Chunk.forceSaveToDisk: data must be a Buffer');
-
-        if (id === null) id = utils.hashFnHex(data);
-
-        // todo: dont zero out the rest of the chunk if it's the last one, save space
-
-        // todo: what if already exists? should we overwrite again or just use it? without integrity check?
-        // todo: important question! because there was a bug related to that, the file was saved under the wrong id and until we cleared the file system we didn't know how to debug it
-        // todo: but if we decide to rewrite here and something goes wrong, we might accidentally rewrite the good version. important for providers not to destroy information, just in case
-        const chunk_file_path = Chunk.getChunkStoragePath(id);
-        if (! fs.existsSync(chunk_file_path)) {
-            fs.writeFileSync(chunk_file_path, data, { encoding: null });
-        }
-
-        return id;
-    }
-
 }
 
-Chunk.UPLOADING_STATUS_CREATED = 'us0';
-Chunk.UPLOADING_STATUS_UPLOADING = 'us1';
-Chunk.UPLOADING_STATUS_FAILED = 'us2';
-Chunk.UPLOADING_STATUS_UPLOADED = 'us99';
-Chunk.DOWNLOADING_STATUS_CREATED = 'ds0';
-Chunk.DOWNLOADING_STATUS_READY_TO_DOWNLOAD = 'ds0';
-Chunk.DOWNLOADING_STATUS_DOWNLOADING = 'ds1';
-Chunk.DOWNLOADING_STATUS_DOWNLOADED = 'ds99';
-Chunk.DOWNLOADING_STATUS_FAILED = 'ds2';
-
 Chunk.init({
     id: { type: Sequelize.DataTypes.STRING, unique: true, primaryKey: true },
     size: { type: Sequelize.DataTypes.INTEGER, allowNull: true },
+    dl_status: { type: Sequelize.DataTypes.STRING, defaultValue: DOWNLOAD_STATUS.NOT_STARTED },
 
-    ul_status: { type: Sequelize.DataTypes.STRING, defaultValue: Chunk.UPLOADING_STATUS_CREATED },
-    dl_status: { type: Sequelize.DataTypes.STRING, defaultValue: Chunk.DOWNLOADING_STATUS_CREATED },
-
-    // allowNull in the next rows because this is only related to uploading and storing, not chunks queued for downloading
+    // TODO: not used, remove
```

Comment:
I think we should use the upload status for chunks as well. The statuses I can think of right away are as follows:
* `in_progress` - needed to avoid attempts to download the same chunk twice
* `failed` - the upload may fail, and the retry logic may be added outside of the storage itself
* `uploaded` - indicates successful upload.

What do you think?

Reply:
As we discussed in the call, using single status for upload and download.